File: FizzBuzz.scala
object FizzBuzz {
  def fizzbuzz(n: Int): String = {
    (n % 3, n % 5) match {
      case (0, 0) => "FizzBuzz"
      case (0, _) => "Fizz"
      case (_, 0) => "Buzz"
      case _ => n.toString
    }
  }

  def main(args: Array[String]): Unit = {
    for (i <- 1 to 100) {
      println(fizzbuzz(i))
    }
  }
}

File: fizzbuzz.ex
defmodule FizzBuzz do
  def fizzbuzz(n) when rem(n, 15) == 0, do: "FizzBuzz"
  def fizzbuzz(n) when rem(n, 3) == 0, do: "Fizz"
  def fizzbuzz(n) when rem(n, 5) == 0, do: "Buzz"
  def fizzbuzz(n), do: Integer.to_string(n)

  def run do
    Enum.each(1..100, fn n ->
      IO.puts(fizzbuzz(n))
    end)
  end
end

FizzBuzz.run()
